pkg/util/helper/index.go

```diff
@@ -18,15 +18,41 @@ package helper
 
 import (
 	"context"
+	"encoding/json"
+	"sync"
 
 	"k8s.io/klog/v2"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
 	workv1alpha1 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha1"
 	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
 )
 
+const (
+	// IndexNameForResourceBindingDependency index name for resourcebinding dependency used for fieldSelector
+	IndexNameForResourceBindingDependency = "resourcebinding.dependency"
+)
+
+var (
+	registerOnce sync.Once
```

Comment:
you don't need this `sync.Once`, add same index twice will return an error.

https://github.com/karmada-io/karmada/blob/6dfe381395c78b935aceaf73a0a873f92b66543c/vendor/k8s.io/client-go/tools/cache/thread_safe_store.go#L123-L135

Reply:
sync.Once is to avoid registering the same index multiple times, we cannot rely on errors

Reply:
But you're only calling it once now, you can add `sync.Once` later when you really start calling it multiple times in the subsequent PR. At least in this PR it's useless. For example, code like `setupControllers()`, `controllerManager.Start()` also only called once, but no one will consider add `sync.Once` for them.

Reply:
refer to this pr: https://github.com/karmada-io/karmada/pull/6204;

Reply:
this pr 6194 in order to add a new index, not for management index